app/controllers/api/v1/slack/user_info_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Api
+  module V1
+    module Slack
+      class UserInfoController < ApplicationController
+        before_action VerifyRequest 
+ 
+        # get all users 
+        def index
+          render json: Users.all
+        end
+
+        # get a single slack user (specifically want :verified and :slack_name)
+        def show
+          user = Users.find_by(email: params[:email])
+          if user 
+            render json: user 
+          else
+            render json: { error: 'No such email record' }, status: :not_found
+          end
+        end
+
+        # update a single slack name by email
+        def update
+          user = User.find(params[:email])
+          if user.update(:slack_name params[:slack_name])
+            render json: user 
+          else
+            render json: { errors: user.errors.full_messages }
+          end
+        end
+      end 
+      
+      class VerifyRequest
+          def self.before(controller)
```

Comment:
Regardless if the logic for the authentication ends up living in here or not, this would still be a private method.  i.e  `def verify_request`.

Instead of `controller`, the parameters for this method would be something that is passed in the `params` hash.  

And that parameter name could perhaps be `credentials`, or something like that.